doc/api/test.md

```diff
@@ -3608,6 +3608,26 @@ test('top level test', async (t) => {
 });
 ```
 
+### `context.waitFor(condition[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `condition` {Function|AsyncFunction} A function that is invoked periodically
+  until it completes successfully or the defined polling timeout elapses. This
+  function does not accept any arguments, and is allowed to return any value.
```

Comment:
Maybe say "an assertion function" or be explicit about the fact that it should throw or reject to signal the polling should retry later.
It's not obvious to me what "completes successfully" means.